packages/editor/src/components/collab-sidebar/comments.js

```diff
@@ -129,9 +123,63 @@ function Thread( {
 	onCommentResolve,
 	isFocused,
 	clearThreadFocus,
+	blockCommentId,
+	onFocusThread,
 } ) {
+	const blocks = useSelect(
+		( select ) => select( blockEditorStore ).getBlocks(),
+		[]
+	);
+
+	const { toggleBlockHighlight } = useDispatch( blockEditorStore );
+
+	// Find first block that have this comment ID - run at component root level.
+	const relatedBlock = useMemo( () => {
+		if ( ! thread.id || ! blocks ) {
+			return null;
+		}
+		const findFirstBlock = ( blockList ) => {
+			for ( const block of blockList ) {
+				if ( block.attributes?.blockCommentId === thread.id ) {
+					return block.clientId;
+				}
+				if ( block.innerBlocks ) {
+					const found = findFirstBlock( block.innerBlocks );
+					if ( found ) {
+						return found;
+					}
+				}
+			}
+			return null;
+		};
+
+		return findFirstBlock( blocks );
+	}, [ thread.id, blocks ] );
```

Comment:
Note for the future (not a blocker): We need to find a better way to derive this value; the `blocks` will update often when the user is typing, which can affect performance.

Unfortunately, we can't do this inside `handleCommentSelect` due to the `useBlockElement` nature.